Text/Mustache/Parser.hs

```diff
@@ -152,12 +152,30 @@ pClosingTag key = do
 {-# INLINE pClosingTag #-}
 
 pKey :: Parser Key
-pKey = (fmap Key . lexeme . label "key") (implicit <|> other)
+pKey =
+  (fmap Key . lexeme . label "key") $
+    implicit <|> do
+      end <- gets closingDel
+
+      -- pKey is not allowed to contain letters from closingDel.
+      --
+      -- This is an undocumented behaviour. We are employing the same strategy as
+      -- found in the package `mustache`.
+      let f x = x `notElem` ('.' : '}' : T.unpack end)
+          lbl = "characters different from those found in \"" ++ (T.unpack end) ++ "\""
```

Comment:
This label is going to look rather strange in error messages. Let's say "key-constituent characters" instead.

Reply:
Sure, I'll change it.